framework/src/constraints/ADNodeElemConstraint.C

```diff
@@ -0,0 +1,78 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ADNodeElemConstraint.h"
+
+// MOOSE includes
+#include "Assembly.h"
+#include "MooseEnum.h"
+#include "MooseMesh.h"
+#include "MooseTypes.h"
+
+#include "libmesh/string_to_enum.h"
+
+InputParameters
+ADNodeElemConstraint::validParams()
+{
+  InputParameters params = NodeElemConstraintBase::validParams();
+  return params;
```

Comment:
missing class description?

Reply:
Do I need one if this class can't be in the input file?

Reply:
ah no you're good